pkg/commands/compute/language_rust.go

```diff
@@ -4,710 +4,314 @@ import (
 	"bufio"
 	"bytes"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"io"
-	"io/fs"
-	"net/http"
 	"os"
 	"os/exec"
 	"path/filepath"
-	"sort"
+	"regexp"
 	"strings"
-	"time"
 
-	"github.com/Masterminds/semver/v3"
-	"github.com/fastly/cli/pkg/api"
 	"github.com/fastly/cli/pkg/config"
 	fsterr "github.com/fastly/cli/pkg/errors"
-	fstexec "github.com/fastly/cli/pkg/exec"
 	"github.com/fastly/cli/pkg/filesystem"
 	"github.com/fastly/cli/pkg/manifest"
 	"github.com/fastly/cli/pkg/text"
 	toml "github.com/pelletier/go-toml"
 )
 
-// RustSourceDirectory represents the source code directory.
-const RustSourceDirectory = "src"
+// RustCompilation is a language specific compilation target that converts the
+// language code into a Wasm binary.
+const RustCompilation = "wasm32-wasi"
 
-// RustManifestName represents the language file for configuring dependencies.
-const RustManifestName = "Cargo.toml"
+// RustCompilationURL is the specification URL for the wasm32-wasi target.
+const RustCompilationURL = "https://doc.rust-lang.org/stable/nightly-rustc/rustc_target/spec/wasm32_wasi/index.html"
 
-// CargoPackage models the package configuration properties of a Rust Cargo
-// package which we are interested in and is embedded within CargoManifest and
-// CargoLock.
-type CargoPackage struct {
-	Name    string `toml:"name" json:"name"`
-	Version string `toml:"version" json:"version"`
-}
+// RustCompilationCommandRemediation is the command to execute to fix the
+// missing compilation target.
+const RustCompilationCommandRemediation = "rustup target add %s --toolchain $ACTIVE_TOOLCHAIN"
```

Comment:
Again, not sure it's necessary to have all of this language-specific validation logic if we are just expecting the user to set the build script.